### PR TITLE
Dtspb 533 new deathcert flow1

### PR DESCRIPTION
**Pull request by meganaslattery**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPB-533

### Change description ###

Flow 1: Normal Death Certificate England and Wales - Citizen journey

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: app/core/steps/EligibilityValidationStep.js [hunk 1]
```diff
@@ -44,33 +45,40 @@ class EligibilityValidationStep extends ValidationStep {
...
-        let eligibilityQuestionsList;
+    previousQuestionsAnswered(req, ctx, currentScreener) {
 
-        if (Object.keys(config.eligibilityQuestionsProbate).includes(fieldKey)) {
-            eligibilityQuestionsList = config.eligibilityQuestionsProbate;
-        } else if (Object.keys(config.eligibilityQuestionsIntestacy).includes(fieldKey)) {
-            eligibilityQuestionsList = config.eligibilityQuestionsIntestacy;
-        }
+        const isIntestacyScreener = Object.keys(config.intestacyScreeners).includes(currentScreener);
+        let screenersList = isIntestacyScreener ? config.intestacyScreeners : config.probateScreeners;
```
Comment (vinceholt) [inline app/core/steps/EligibilityValidationStep.js:51]: Could potentially reduce conditional logic by moving out getScreeners to another method

File: app/core/steps/EligibilityValidationStep.js [hunk 2]
```diff
@@ -44,33 +45,40 @@ class EligibilityValidationStep extends ValidationStep {
...
+            if (isIntestacyScreener) {
+                screenersList = deathCertificateNotInEnglish ? config.intestacyScreenersDeathCertificateNotInEnglish : config.intestacyScreenersDeathCertificateInEnglish;
+            } else {
+                screenersList = deathCertificateNotInEnglish ? config.probateScreenersDeathCertificateNotInEnglish : config.probateScreenersDeathCertificateInEnglish;
             }
+        }
 
-            if (!allPreviousEligibilityQuestionsAnswered) {
-                return false;
+        let allPreviousEligibilityQuestionsAnswered = true;
```
Comment (vinceholt) [inline app/core/steps/EligibilityValidationStep.js:68]: Could potentially encapsulate areAllPreviousEligibilityQuestionsAnswered into another method

File: app/middleware/multipleApplications.js [hunk 3]
```diff
@@ -58,13 +59,19 @@ const createNewApplication = (req, res, formdata, formData, result, next) => {
...
-const allEligibilityQuestionsPresent = (formdata) => {
+const allEligibilityQuestionsPresent = (formdata, featureToggles) => {
     let allQuestionsPresent = true;
 
     if (formdata.screeners && formdata.screeners.left) {
-        let eligibilityQuestionsList = config.eligibilityQuestionsProbate;
-        if (formdata.screeners.left === 'optionNo') {
-            eligibilityQuestionsList = config.eligibilityQuestionsIntestacy;
+        let eligibilityQuestionsList = formdata.screeners.left === 'optionNo' ? config.intestacyScreeners :config.probateScreeners;
+
```
Comment (vinceholt) [inline app/middleware/multipleApplications.js:67]: Missing a space after the colon

File: app/core/steps/EligibilityValidationStep.js (hunk 1 above)
Comment (meganaslattery) [reply app/core/steps/EligibilityValidationStep.js]: Done with my last commit :)

**Review by vinceholt — APPROVED** [review]

**Review by elvianihuseyin — APPROVED** [review]
LGTM

**Pull request merged**